shared/util/quit-helper.desktop.js

```diff
@@ -3,18 +3,21 @@ import {ipcRenderer, ipcMain, app} from 'electron'
 import {quit} from '../../desktop/app/ctl'
 import {hideDockIcon} from '../../desktop/app/dock-icon'
 
-export type Context = {type: 'uiWindow'} | {type: 'mainThread'} | {type: 'quitButton'}
+export type Context = 'uiWindow' | 'mainThread' | 'quitButton' | 'beforeQuit'
```

Comment:
I think we should add complexity as we need it when we understand what
we'll need (if we need it) and not preemptively make it more complex in
case we might need it

On Friday, July 15, 2016, Marco Munizaga <notifications@github.com> wrote:

> In shared/util/quit-helper.desktop.js
> <https://github.com/keybase/client/pull/3469#discussion_r71045261>:
>
> > @@ -3,18 +3,21 @@ import {ipcRenderer, ipcMain, app} from 'electron'
> >  import {quit} from '../../desktop/app/ctl'
> >  import {hideDockIcon} from '../../desktop/app/dock-icon'
> >
> > -export type Context = {type: 'uiWindow'} | {type: 'mainThread'} | {type: 'quitButton'}
> > +export type Context = 'uiWindow' | 'mainThread' | 'quitButton' | 'beforeQuit'
>
> Not sure about this, the reason I did that is so we can add stuff to them
> later if we needed to, this makes it a bit harder
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/3469/files/696a85227ca0f7720ffeeea0c9eb4a149973b035#r71045261>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfhUxdHWP5QW1ca8fYNS67nu0G7a00tks5qV__RgaJpZM4JMsjQ>
> .
>